tests/php/Core/Injector/InjectorTest.php

```diff
@@ -815,6 +816,28 @@ public function testConvertServicePropertyOnCreate()
         $this->assertInstanceOf(OtherTestObject::class, $item->property->property);
     }
 
+    /**
+     * @dataProvider provideConvertServicePropertyBackTicks
+     */
+    public function testConvertServicePropertyBackTicks($value, $expected)
+    {
+        Environment::setEnv('INJECTOR_TEST_CSP_A', 'ABC');
+        Environment::setEnv('INJECTOR_TEST_CSP_B', 'DEF');
+        $actual = Injector::inst()->convertServiceProperty($value);
+        $this->assertSame($expected, $actual);
+    }
+
+    public function provideConvertServicePropertyBackTicks()
+    {
+        return [
+            ['`INJECTOR_TEST_CSP_A`', 'ABC'],
+            ['`INJECTOR_TEST_CSP_A`:`INJECTOR_TEST_CSP_B`', 'ABC:DEF'],
```

Comment:
```suggestion
            ['`INJECTOR_TEST_CSP_A`:`INJECTOR_TEST_CSP_B`', 'ABC:DEF'],
            ['`INJECTOR_TEST_CSP_A` some text `INJECTOR_TEST_CSP_B`', 'ABC some text DEF'],
            ['`INJECTOR_TEST_CSP_A``INJECTOR_TEST_CSP_B`', 'ABCDEF'],
```
Adding in an example with no characters between, and with some non-colon characters between the two vars.

Reply:
What would be the intended outcome of ```'`INJECTOR_TEST_CSP_C``INJECTOR_TEST_CSP_D`'```? I'm guessing null but we should probably test that case as well.

Reply:
Can you please also add a test case with three variables? Just to be sure it's not just grabbing the first and last.

Reply:
Updated